hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementAsProxyLazinessInterceptor.java

```diff
@@ -149,15 +151,17 @@ protected Object handleRead(Object target, String attributeName, Object value) {
 					if ( writtenValues != null ) {
 						// here is the replaying of the explicitly set values we prepared above
 						for ( String writtenFieldName : writtenFieldNames ) {
-							final int size = entityPersister.getNumberOfAttributeMappings();
-							for ( int index = 0; index < size; index++ ) {
+							for ( int index = 0; index < numberOfAttributeMappings; index++ ) {
 								final String name = entityPersister.getAttributeMapping( index ).getAttributeName();
 								if ( writtenFieldName.contains( name ) ) {
```

Comment:
Shouldn't this rather be this?

```suggestion
								if ( writtenFieldName.equals( name ) ) {
```

Or maybe instead of iterating all attributes mappings, use `entityPersister.findAttributeMapping( writtenFieldName )` directly?

Reply:
we need the index of the attribute

Reply:
Yeah, but the attribute offers the index already through `getStateArrayPosition()`

Reply:
@beikov what do you think of my last change?